### PR TITLE
chore: sign psbt laboratory

### PR DESCRIPTION
**Pull request by zoruka**
# Description

Adds PSBT creation util on laboratory.

## Type of change

- [x] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-xxx
For GH issues: closes #...

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [ ] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [ ] My changes generate no new warnings
- [ ] I have reviewed my own code
- [ ] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: apps/laboratory/src/utils/BitcoinUtil.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+          script: payment.output,
+          value: BigInt(utxo.value)
+        }
+      })
+    }
+
+    psbt.addOutput({
+      address: params.recipientAddress,
+      value: BigInt(params.amount)
+    })
```
Comment (tomiir) [inline apps/laboratory/src/utils/BitcoinUtil.ts:42]: we'll probably need this internally on the wallet standard side of things right?

File: apps/laboratory/src/utils/BitcoinUtil.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    for (const utxo of params.utxos) {
+      psbt.addInput({
+        hash: utxo.txid,
+        index: utxo.vout,
+        witnessUtxo: {
+          script: payment.output,
+          value: BigInt(utxo.value)
+        }
```
Comment (tomiir) [inline apps/laboratory/src/utils/BitcoinUtil.ts:35]: is this always a witness utxo? doesn't it depend on the key type?

File: apps/laboratory/src/utils/BitcoinUtil.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      `https://mempool.space${isTestnet ? '/testnet' : ''}/api/address/${address}/utxo`
+    )
+
+    return await response.json()
+  },
+
+  async getFeeRate() {
+    const defaultFeeRate = 2
+    try {
+      const response = await fetch('https://mempool.space/api/v1/fees/recommended')
```
Comment (tomiir) [inline apps/laboratory/src/utils/BitcoinUtil.ts:79]: @geekbrother any way we can have this in blockchain api ?

**Review by dav3yblaz3 — APPROVED** [review]
let us succeed

File: apps/laboratory/src/utils/BitcoinUtil.ts (hunk 2 above)
Comment (zoruka) [reply apps/laboratory/src/utils/BitcoinUtil.ts:35]: What do you mean by key type?

File: apps/laboratory/src/utils/BitcoinUtil.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    return {
+      psbt: psbt.toBase64(),
+      signInputs: [],
+      broadcast: false
+    }
+  },
+
+  async getUTXOs(address: string, networkId: CaipNetworkId): Promise<BitcoinUtil.UTXO[]> {
```
[the remark this answers is not in the document]
Comment (zoruka) [reply apps/laboratory/src/utils/BitcoinUtil.ts:65]: what would be a optimized method?

File: apps/laboratory/src/utils/BitcoinUtil.ts (hunk 1 above)
Comment (zoruka) [reply apps/laboratory/src/utils/BitcoinUtil.ts:42]: Yeah you are correct, we will probably need that in WalletStandard to create the transfer functionality. I think we will need to get it differently as we are relying into an API to get the UTXO and fees, so we need to check which would be the best approach to have this internally.

File: apps/laboratory/src/utils/BitcoinUtil.ts (hunk 2 above)
Comment (tomiir) [reply apps/laboratory/src/utils/BitcoinUtil.ts:35]: https://www.hiro.so/blog/understanding-the-differences-between-bitcoin-address-formats-when-developing-your-app

File: apps/laboratory/src/utils/BitcoinUtil.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (tomiir) [reply apps/laboratory/src/utils/BitcoinUtil.ts:65]: you can optimally select utxos to minimize change, we can probs use trezors lib to do this

File: apps/laboratory/src/utils/BitcoinUtil.ts (hunk 2 above)
Comment (zoruka) [reply apps/laboratory/src/utils/BitcoinUtil.ts:35]: maybe we should check the address before using `p2wpkh`?

File: apps/laboratory/src/utils/BitcoinUtil.ts (hunk 3 above)
Comment (dav3yblaz3) [reply apps/laboratory/src/utils/BitcoinUtil.ts:79]: yes team please let's get someone assigned to this

**Comment by dav3yblaz3** [comment]
thank you for all the efforts greatest team ever we can get this we try for now

**Pull request merged**